drivers/include/lc709203f.h

```diff
@@ -0,0 +1,347 @@
+ * @brief       Device driver interface for the LC709203F Battery Fuel Gauge
+ *
+ * @author      Steffen Robertz <steffen.robertz@rwth-aachen.de>
+ * @author      Josua Arndt <jarndt@ias.rwth-aachen.de>
+ */
+
+#ifndef LC709203F_H
+#define LC709203F_H
+
+#ifdef __cplusplus 
+extern "C" {
+#endif
+
+#include "periph/i2c.h"
+#include "periph/gpio.h"
+
+typedef enum {
+    AUTO_MODE = 0x0,
+    CHARGE_MODE = 0x1,
+    DISCHARGE_MODE = 0xffff
+} lc709203f_current_direction_t;
+
+typedef enum {
+    BAT_PROFILE_1 = 0,
+    BAT_PROFILE_2 = 1
+} lc709203f_battery_profile_t;
+
+typedef enum {
+    OPERATIONAL_MODE = 1,
+    SLEEP_MODE = 2
+} lc709203f_power_mode_t;
+
+typedef enum {
+    I2C_MODE = 0,
+    THERMISTOR_MODE = 1
+} lc709203f_temp_obtaining_mode_t;
+
+typedef void (*gauge_cb_t)(void *arg);
+
+typedef struct {
+    gpio_t alarm_pin;
+    i2c_t bus;
+    uint8_t addr;
+} lc709203f_params_t;
+
+enum {
+    LC709203F_OK    = 0,  /**< all went as expected */
+    LC709203F_NOI2C = -1, /**< error using the I2C bus */
+    LC709203F_CELL_TEMP_INVALID = -2 /**< Cell temp invalid */
+};
+
+typdef struct {
+    i2c_t bus;                  /**< I2C bus to use */
+    uint8_t addr;               /**< I2C Address of fuel gauge */
+    uint8_t scale;              /**< scale for returned values */
+    lc709203f_params_t params;  /**< param struct with static settings etc*/
+    lc709203f_cb_t cb;          /**< callback method*/
+    void *arg;                  /**< additional arguments for the callback method*/
+} lc709203f_t
+
+/*CRC I2C-Addresses for Read/write*/
+#define LC709203F_I2C_READ              0x17
+#define LC709203F_I2C_WRITE             0x16
+/*Defines for Registers*/
+#define LC709203F_REG_CELL_VOLTAGE      0x09
+#define LC709203F_REG_RSOC              0x0d
+#define LC709203F_REG_ITE               0x0f
+#define LC709203F_REG_ID                0x11
+#define LC709203F_REG_CELL_TEMP         0x08
+#define LC709203F_REG_STATUS            0x16
+#define LC709203F_REG_POWER_MODE        0x15
+#define LC809203F_REG_ALARM_VOLTAGE     0x14
+#define LC709203F_REG_ALARM_RSOC        0x13
+#define LC709203F_REG_CHANGE_PARAMETER  0x12
+#define LC709203F_REG_APT               0x0c
+#define LC709203F_REG_APA               0x0b
+#define LC709203F_REG_CURRENT_DIRECTION 0x0a
+#define LC709203F_REG_THERMISTOR        0x06
+#define LC709203F_REG_BEFORE_RSOC       0x04
+#define LC709203F_REG_INITIAL_RSOC      0x07
+
+/*Define CRC Polynomial*/
+#define LC709203F_CRC_POLYNOMIAL        0x07
+/**
+ * @brief initializes the sensor and i2c
+ *
+ * @param[in] *dev      pointer to lc709203f_t struct containing the i2c device and the address
+ * @param[in] *params   pointer to lc709203f_param_t struct containing the interrupt pin and callback
+ *
+ * @return              LC709203F_OK on a working initialization
+ * @return              LC709203F_NOI2C on undefined i2c device given in periph_conf
+  */
+int lc709203f_init(lc709203f_t *dev, lc709203f_params_t *params);
```

Comment:
add `const` to params here, and where applicable in the other methods below (to actually match lc709203f.c)